src/device/intrinsics/math.jl

```diff
@@ -310,7 +308,7 @@ Base.expm1(x::Float16) = Float16(CUDA.expm1(Float32(x)))
 
 @device_override Base.fma(x::Float64, y::Float64, z::Float64) = ccall("extern __nv_fma", llvmcall, Cdouble, (Cdouble, Cdouble, Cdouble), x, y, z)
 @device_override Base.fma(x::Float32, y::Float32, z::Float32) = ccall("extern __nv_fmaf", llvmcall, Cfloat, (Cfloat, Cfloat, Cfloat), x, y, z)
-# @device_override Base.fma(x::Float16, y::Float16, z::Float16) = @asmcall("fma.rnd.f16 \$0, \$1, \$2, \$3", "=h,h,h,h", Float16, Tuple{Float16, Float16, Float16}, x, y, z)
+@device_override Base.fma(x::Float16, y::Float16, z::Float16) = @asmcall("fma.rn.f16 \$0, \$1, \$2, \$3;", "=h,h,h,h", true, Float16, Tuple{Float16, Float16, Float16}, x, y, z)
```

Comment:
Is there no LLVM intrinsics for this? Does ` @llvm.fma.f16` expand to the right thing?

Reply:
Looks like it does, and it passes the new tests I've written for `CUDA.fma`. I'm not entirely sure which one is more performant (or if there's even a difference) but I've changed it for now.